pallets/capacity/src/lib.rs

```diff
@@ -477,6 +715,36 @@ impl<T: Config> Pallet<T> {
 		Ok(capacity)
 	}
 
+	fn increase_stake_and_issue_boost_capacity(
+		staker: &T::AccountId,
+		staking_details: &mut StakingDetails<T>,
+		target: &MessageSourceId,
+		amount: &BalanceOf<T>,
+	) -> Result<BalanceOf<T>, DispatchError> {
+		staking_details.deposit(*amount).ok_or(ArithmeticError::Overflow)?;
+		Self::set_staking_account_and_lock(staker, staking_details)?;
+
+		// get the capacity generated by a Provider Boost
+		let capacity = Self::capacity_generated(T::RewardsProvider::capacity_boost(*amount));
+
+		let mut target_details = Self::get_target_for(staker, target).unwrap_or_default();
+
+		target_details.deposit(*amount, capacity).ok_or(ArithmeticError::Overflow)?;
+		Self::set_target_details_for(staker, *target, target_details);
+
+		let mut capacity_details = Self::get_capacity_for(target).unwrap_or_default();
+		capacity_details.deposit(amount, &capacity).ok_or(ArithmeticError::Overflow)?;
+		Self::set_capacity_for(*target, capacity_details);
+
+		let era = Self::get_current_era().era_index;
+		Self::upsert_boost_history(staker, era, *amount, true)?;
+
+		let reward_pool_total = CurrentEraProviderBoostTotal::<T>::get();
+		CurrentEraProviderBoostTotal::<T>::set(reward_pool_total.saturating_add(*amount));
```

Comment:
My understanding is that a user will only be rewarded for completing a full era, meaning they won’t receive rewards for the era in which they initially stake. However, it seems that when a user stakes, their stake affects the total amount that everyone can stake. 

Does this imply that new stakers reduce the rewards for those who have already staked?

Reply:
The reward pool total is set at the _beginning_ of each era (in `start_new_reward_era_if_needed`).  So the reward is not changed until the _following_ era, because all staking changes don't go into effect until they've been that way for the entire era.